src/sGLOH2.cpp

```diff
@@ -55,38 +55,43 @@ void sGLOH2::compute(const cv::Mat& image, std::vector<cv::KeyPoint>& keypoints,
 
 cv::Mat sGLOH2::compute_sGLOH_single(const cv::Mat& patch) {
     // Initialize the descriptor as a single-row matrix of zeros.
-    // The type of the matrix is CV_32F, which means that it contains 32-bit floating-point numbers.
     cv::Mat descriptor = cv::Mat::zeros(1, 0, CV_32F);
 
-    // Compute the size of the regions into which the patch will be divided.
-    // Each region will be a square with side length equal to the number of rows in the patch divided by m.
-    int region_size = patch.rows / m;
-
-    // Compute the histogram for each region.
-    // The patch is divided into m*m regions, and a histogram is computed for each one.
-    for (int r = 0; r < m; r++) {
-        for (int d = 0; d < m; d++) {
-            // Compute the region.
-            // The region is a square with top-left corner at (r*region_size, d*region_size) and side length region_size.
-            cv::Rect region(r * region_size, d * region_size, region_size, region_size);
-
-            // Compute the histogram for the region.
-            // The computeHistogram function computes a histogram of gradient orientations for the region.
-            cv::Mat histogramMat = computeHistogram(patch(region), m);
-
-            // Concatenate the histogram to the descriptor.
-            // The hconcat function horizontally concatenates the histogram to the descriptor.
-            // This means that the histogram is added as new columns at the end of the descriptor.
-            cv::hconcat(descriptor, histogramMat, descriptor);
+    // Compute the radius of the circles.
+    int radius = patch.rows / (2 * m);  // The radius is half the patch size divided by m.
+
+    // Compute the histogram for each circular region.
+    for (int i = 0; i < m; i++) {
```

Comment:
There should be n*m total regions right?

Reply:
I think we need something where we iterate through each m for each n -> ie each m sector of each n ring around the keypoint

Reply:
"The sGLOH descriptor grid is made up of n × m regions...defined by n rings centered on the keypoint, each containing
m sectors, equally distributed along m directions"

Reply:
Not sure still working on getting it right